docs/DeveloperGuide.adoc

```diff
@@ -314,16 +370,144 @@ _{More to be added}_
 == Use Cases
 
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+[discrete]
+=== Use case: Add person
+Guarantees: +
+
+1. Person is added even if input fields might be invalid (see 1a).
+
+*MSS*
+
+1.  User requests to add a person.
+2.  AddressBook adds the person.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Either of the given NRIC, contact number, or email address is invalid.
++
+[none]
+** 1a1. AddressBook adds the person into address book.
+** 1a1. AddressBook shows a warning.
+
+* 1b. Duplicate profile is added.
++
+[none]
+** 1b1. AddressBook shows an error message and will attempt to merge the profile.
++
+
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Edit person
+Guarantees: +
+
+1. Person is editted even if input fields might be invalid (see 3a).
+
+*MSS*
+
+1.  User requests to find a person
+2.  AddressBook shows details of the person
+3.  User requests to edit that specific person.
+4.  AddressBook edits the person.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Person is not found
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case resumes at step 1.
+
+* 3a. Either of the given NRIC, contact number, or email address is invalid.
++
+[none]
+** 3a1. AddressBook adds the person into address book.
+** 3a2. AddressBook shows a warning.
++
+
+* *a. At any time, User requests to cancel the edit.
++
+[none]
+** *a1. Address books requests to confirm the cancellation.
+** *a1. User confirms the cancellation.
+Use case ends.
+
+[discrete]
+=== Use case: Delete contact
+
+*MSS*
+
+1.  User requests to list contacts
+2.  AddressBook shows a list of contacts
+3.  User requests to delete a specific contact in the list
+4.  AddressBook deletes the contact
+5.  Contact appears in the recycling bin
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
```

Comment:
Maybe you could display an error message to show that list is empty, so that user would not mistakenly think that the contacts is successfully deleted.